conformance/utils/kubernetes/helpers.go

```diff
@@ -108,14 +108,21 @@ func gwcMustBeAccepted(t *testing.T, c client.Client, timeoutConfig config.Timeo
 	return controllerName
 }
 
-// GatewayMustHaveLatestConditions will fail the test if there are
-// conditions that were not updated
-func GatewayMustHaveLatestConditions(t *testing.T, gw *v1beta1.Gateway) {
+// GatewayMustHaveLatestConditions waits until the specified Gateway has
+// the latest conditions
+func GatewayMustHaveLatestConditions(t *testing.T, timeoutConfig config.TimeoutConfig, gw *v1beta1.Gateway) {
 	t.Helper()
 
-	if err := ConditionsHaveLatestObservedGeneration(gw, gw.Status.Conditions); err != nil {
-		t.Fatalf("Gateway %v", err)
-	}
+	waitErr := wait.PollImmediate(1*time.Second, timeoutConfig.GatewayObservedGenerationBumpTimeout, func() (bool, error) {
+		if err := ConditionsHaveLatestObservedGeneration(gw, gw.Status.Conditions); err != nil {
+			t.Log("Gateway", err)
```

Comment:
Can you add a more descriptive error message here? (I know this is basically the same as what already existed, but an improvement here would be nice).